src/middlewared/middlewared/plugins/device_/device_info_freebsd.py

```diff
@@ -0,0 +1,103 @@
+import re
+
+from bsd import geom, devinfo
+
+from .device_info_base import DeviceInfoBase
+from middlewared.service import private, Service
+
+
+RE_DISK_NAME = re.compile(r'^([a-z]+)([0-9]+)$')
+
+
+class DeviceService(Service, DeviceInfoBase):
+
+    async def get_disks(self):
+        disks = {}
+        klass = await self.middleware.call('device.retrieve_disk_geom_klass')
+        if not klass:
+            return disks
+        for g in klass.geoms:
+            # Skip cd*
+            if g.name.startswith('cd'):
+                continue
+            disks[g.name] = await self.get_disk_details(self.disk_default.copy(), g)
+        return disks
+
+    @private
+    def retrieve_disk_geom_klass(self):
```

Comment:
I dont quite understand why we have to call it klass instead of class, its the full method name and not just the "class" keyword. But just pointing that out, no need to change ;)